File: app/src/main/java/com.caique.everis.testeandroid/data/local/AppDataBase.java
package com.caique.everis.testeandroid.data.local;

import android.arch.persistence.room.Database;
import android.arch.persistence.room.Room;
import android.arch.persistence.room.RoomDatabase;
import android.content.Context;

import com.caique.everis.testeandroid.data.local.dao.LoginDao;
import com.caique.everis.testeandroid.model.Login;

@Database(entities = {Login.class}, version = 1, exportSchema = false)
public abstract class AppDataBase extends RoomDatabase {

    private static final String DATABASE_NAME = "songstats";
    private static AppDataBase instance;

    public static AppDataBase getAppDatabase(Context context) {
        if (instance == null) {
            synchronized (AppDataBase.class) {
                instance = Room.databaseBuilder(context.getApplicationContext(),
                        AppDataBase.class,
                        DATABASE_NAME)
                        .allowMainThreadQueries()
                        .build();
            }
        }
        return instance;
    }

    public abstract LoginDao loginDao();
}

File: Bank/app/src/main/java/com/tata/bank/database/AppDatabase.kt
package com.tata.bank.database

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase

@Database(entities = [LoginCredentialEntity::class], version = 1, exportSchema = false)
abstract class AppDatabase : RoomDatabase() {

    abstract fun credentialsDao(): LoginCredentialDao

    companion object {
        @Volatile
        private var instance: AppDatabase? = null

        operator fun invoke(context: Context) = instance
            ?: synchronized(this) {
                instance
                    ?: buildDatabase(
                        context
                    ).also { instance = it}
            }

        private fun buildDatabase(context: Context) = Room.databaseBuilder(context.applicationContext,
            AppDatabase::class.java, "bank_database")
            .build()
    }
}